src/robotFactory.js

```diff
@@ -1,10 +1,86 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.weight = weight;
 
-class FlyingRobot {}
+    const {
+      x = 0,
+      y = 0,
+    } = coords;
 
-class DeliveryDrone {}
+    this.coords = {
+      x: x,
+      y: y,
+    };
+
+    this.chipVersion = chipVersion;
+  };
+
+  getInfo() {
+    const nameInf = this.name;
+    const version = this.chipVersion;
+    const weightInf = this.weight;
+
+    return `Robot: ${nameInf}, Chip version: ${version}, Weight: ${weightInf}`;
+  }
```

Comment:
```suggestion
  getInfo() {
    return `Robot: ${this.name}, `
      + `Chip version: ${this.chipVersion}, `
      + `Weight: ${this.weight}`;
  }
```
не потрібно створювати лишні змінні тут